### ISSUE TITLE
Compute learning curves for NER models

### ISSUE THREAD

**Issue by FrancescoCasalegno** [BlueBrain/Search#601]
## Context

- Assume that we finished collecting NER ground-truth annotations from our expert.
- The entity types of interest are `BRAIN_REGION`, `CELL_COMPARTMENT`, `CELL_TYPE`, `GENE`, `ORGANISM`.
- For this issue, use SpaCy to train the models.

## Actions

- [ ] Generate two plots like the ones in the image, using as x-axis (1) the "train set size" and (2) the "train set support". Choose to plot e.g. 5 points (size 1x, 2x, 4x, 8x, 16x).  
       ![Screen Shot 2022-06-22 at 23 43 11](https://user-images.githubusercontent.com/17013890/175144019-98d9fd28-5b7b-445c-8650-9336bcb334bb.png)
- [ ] Compute curves for all metrics of interest: `precision`, `recall`, `f1-score`.
- [ ] For each curve, report also the slope of a fitted line (should be done in the log-log scale, not in the lin-lin[^1]) e.g. using `sklearn`'s [`LinearRegression`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html). Note that the learning curve may "saturate" at a certain point, so it may be better to fit the line on the first 3-4 points only (or use some sort of robust regression), see following example:
       ![Screen Shot 2022-06-23 at 00 00 58](https://user-images.githubusercontent.com/17013890/175157461-9e582c1e-eb56-469c-9b32-d4b7d5b0cc95.png)
- [ ] Save a CSV file with the results, so that these curves can be recomputed in the future.

[^1]: If the relation looks linear in the log-log space (which should be our case), it means that `log y = a log x  + b`. But then in the lin-lin space we have `y = b  x ^ a `, i.e. a [Power Law](https://en.wikipedia.org/wiki/Power_law).

**Comment by jankrepl** [comment]
See below all the results we generate.

*  The first row of plots represents linear regression fitted on `log(y), log(y)` and corresponding 95% confidence intervals (`sns.regplot`)
* The second row shows how the predicted relationship looks in the "x, y" space
* The last row shows for each target error how many samples it would take to get to that error


# model = "spacy", metric = "precision", x = "n_train"
![Screenshot 2022-06-28 at 16 44 13](https://user-images.githubusercontent.com/18519371/176210213-a5e4169b-26b6-42bd-91e1-b05ef11e1f30.png)

# model = "spacy", metric = "recall",  x = "n_train"
![Screenshot 2022-06-28 at 16 44 22](https://user-images.githubusercontent.com/18519371/176210269-d79b9c3f-cf1f-4a54-8ed1-2712c14f2a29.png)

# model = "spacy", metric = "f1-score",  x = "n_train"
![Screenshot 2022-06-28 at 16 44 32](https://user-images.githubusercontent.com/18519371/176210306-774c87eb-9084-4ffb-9aa4-5de6e147da26.png)

# model = "hf", metric = "precision",  x = "n_train"
![Screenshot 2022-06-28 at 16 44 53](https://user-images.githubusercontent.com/18519371/176210335-d2872435-0b06-4031-bba2-f9f1982589e3.png)

# model = "hf", metric = "recall",  x = "n_train"
![Screenshot 2022-06-28 at 16 45 09](https://user-images.githubusercontent.com/18519371/176210361-c17c9880-958e-4517-a0ac-e38e89d269fa.png)

# model = "hf", metric = "f1-score",  x = "n_train"
![Screenshot 2022-06-28 at 16 45 20](https://user-images.githubusercontent.com/18519371/176210389-2adbe414-7e8d-4ea1-ac78-a26bc7d8be60.png)


# model = "spacy", metric = "precision", x = "support"
![Screenshot 2022-06-28 at 16 52 55](https://user-images.githubusercontent.com/18519371/176211086-e2ae31cd-783e-414d-8dcd-130fcfb00f68.png)

# model = "spacy", metric = "recall",  x = "support"
![Screenshot 2022-06-28 at 16 53 04](https://user-images.githubusercontent.com/18519371/176211135-cd9e4d41-05cc-4e88-8f6c-d44ba40fba1e.png)

# model = "spacy", metric = "f1-score",  x = "support"
![Screenshot 2022-06-28 at 16 53 15](https://user-images.githubusercontent.com/18519371/176211189-5f671824-ae47-4b4d-8a5e-aafeed70d97f.png)

# model = "hf", metric = "precision",  x = "support"
![Screenshot 2022-06-28 at 16 53 28](https://user-images.githubusercontent.com/18519371/176211240-3971be9d-2048-49b3-a649-65ddc49860b1.png)

# model = "hf", metric = "recall",  x = "support"
![Screenshot 2022-06-28 at 16 53 35](https://user-images.githubusercontent.com/18519371/176211292-07ec7954-40c8-4418-8910-5bdd9d53e87f.png)

# model = "hf", metric = "f1-score",  x = "support"
![Screenshot 2022-06-28 at 16 53 44](https://user-images.githubusercontent.com/18519371/176211343-25f6aea6-d675-4411-95f9-8574dee4cf58.png)

**Comment by jankrepl** [comment]
Note that all the (detailed) data that were used to generate the above plots are here
[output_github.csv](https://github.com/BlueBrain/Search/files/9091415/output_github.csv)
